inc/supplier.class.php

```diff
@@ -459,26 +459,32 @@ function showInfocoms() {
                $linkfield = 'consumableitems_id';
             }
 
+            if ($itemtype == 'Item_DeviceControl') {
+               $query .= "INNER JOIN `glpi_devicecontrols`
+                           ON (`glpi_items_devicecontrols`.`devicecontrols_id`=`glpi_devicecontrols`.`id`)";
+               $linktype = 'DeviceControl';
+               $linkfield = 'devicecontrols_id';
+            }
+
             $linktable = getTableForItemType($linktype);
+            $link_item = new $linktype();
 
+            $query = str_replace('NAME_FIELD', $link_item->getNameField(), $query);
             $query .= "WHERE `glpi_infocoms`.`itemtype` = '$itemtype'
                              AND `glpi_infocoms`.`suppliers_id` = '$instID'".
                              getEntitiesRestrictRequest(" AND", $linktable) ."
                        ORDER BY `glpi_infocoms`.`entities_id`,
-                                `$linktable`.`name`";
+                                `$linktable`.`" . $link_item->getNameField() . "`";
 
             $result_linked = $DB->query($query);
             $nb            = $DB->numrows($result_linked);
 
-            // Set $linktype for link to search engine pnly
-            if (($itemtype == 'SoftwareLicense')
-                && ($nb > $_SESSION['glpilist_limit'])) {
-               $linktype  = 'Software';
-               $linkfield = 'softwares_id';
-            }
-            $link_item = new $linktype();
-
             if ($nb > $_SESSION['glpilist_limit']) {
+               if ($itemtype == 'SoftwareLicense') {
+                  // Set $linktype for link to search engine only
```

Comment:
you keep this part finally ?
Since 9.1, we have a menu entry for SoftwareLicences